File: libs/src/main/java/org/zsy/libs/ext/ViewExt.java
package org.zsy.libs.ext;

import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;

public class ViewExt {


    public static int minClickPeriod = 1000;
    private static long lastClickMs = -1L;

    public static void setMinClickPeriod(int minClickPeriod) {
        ViewExt.minClickPeriod = minClickPeriod;
    }

    public static void addBtn(LinearLayout ll, String btnText, View.OnClickListener clickListener) {
        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
        Button button = new Button(ll.getContext());
        button.setAllCaps(false);
        button.setText(btnText);
        button.setOnClickListener(clickListener);
        ll.addView(button, params);

    }

    public static void setOnSlowlyClickListener(View view, final View.OnClickListener clickListener) {
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                long currentMs = System.currentTimeMillis();
                if (currentMs - lastClickMs > minClickPeriod) {
                    lastClickMs = currentMs;
                    clickListener.onClick(v);
                }
            }
        });
    }


}

File: libs/src/main/java/org/zsy/libs/ext/ViewExt.kt
package org.zsy.libs.ext

import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.LinearLayout


private var lastClickMs = -1L

fun LinearLayout.addButton(text: String, onclick: (view: View) -> Unit) {
    val layoutParams = ViewGroup.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.WRAP_CONTENT
    )
    val button = Button(context).apply {
        isAllCaps = false
        this.text = text
        setOnClickListener(onclick)
    }

    addView(button, layoutParams)


}


fun View.setOnSlowlyClickListener(
        minPeriod: Int = 1000,
        onclick: (view: View) -> Unit
) {
    setOnClickListener {
        val currentMs = System.currentTimeMillis()
        if (currentMs - lastClickMs >= minPeriod) {
            lastClickMs = currentMs
            onclick(it)
        }
    }
}